packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+                const { id } = child.props;
+                return (
+                    <TabTitle
+                        {...child.props}
+                        parentId={this.props.id}
+                        onClick={this.getTabClickHandler(id)}
+                        selected={id === selectedTabId}
+                    />
+                );
+            } else {
+                // TabTitle renders an <li> so let's do the same here
+                return <li>{child}</li>;
+            }
+        });
+
+        const tabPanels = this.getTabChildren()
+            .filter(this.props.renderActiveTabPanelOnly ? (tab) => tab.props.id === selectedTabId : () => true)
+            .map(this.renderTabPanel);
+
+        const tabIndicator = (
+            <div className="pt-tab-indicator-wrapper" style={indicatorWrapperStyle}>
+                <div className="pt-tab-indicator" />
+            </div>
+        );
+
+        const classes = classNames(Classes.TABS, { "pt-vertical": this.props.vertical }, this.props.className);
+        return (
+            <div className={classes}>
+                <div
+                    className={Classes.TAB_LIST}
+                    onKeyDown={this.handleKeyDown}
+                    onKeyPress={this.handleKeyPress}
+                    ref={this.handleTabRef}
+                    role="tablist"
+                >
+                    {this.props.animate ? tabIndicator : undefined}
+                    {tabTitles}
+                </div>
+                {tabPanels}
+            </div>
+        );
+    }
+
+    public componentDidMount() {
+        this.moveSelectionIndicator();
+    }
+
+    public componentWillReceiveProps({ selectedTabId }: ITabs2Props) {
+        if (selectedTabId !== undefined) {
+            // keep state in sync with controlled prop, so state is canonical source of truth
+            this.setState({ selectedTabId });
+        }
+    }
+
+    public componentDidUpdate(_: ITabs2Props, prevState: ITabs2State) {
+        if (this.state.selectedTabId !== prevState.selectedTabId) {
+            this.moveSelectionIndicator();
+        }
+    }
+
+    /** Filters this.props.children to only `<Tab>`s */
+    private getTabChildren() {
+        return React.Children.toArray(this.props.children).filter(isTab) as TabElement[];
+    }
+
+    /** Queries root HTML element for all `.pt-tab`s with optional filter selector */
+    private getTabElements(subselector = "") {
+        if (this.tabElement == null) {
+            return [] as Elements;
+        }
+        return this.tabElement.queryAll(TAB_SELECTOR + subselector);
+    }
+
+    private getTabClickHandler(newTabId: TabId) {
+        return () => {
+            const isControlled = this.props.selectedTabId !== undefined;
+            const { selectedTabId } = isControlled ? this.props : this.state;
+            if (newTabId !== selectedTabId) {
+                safeInvoke(this.props.onChange, newTabId, selectedTabId);
```

Comment:
nit: We're pretty inconsistent syntactically with `Utils.safeInvoke` vs `safeInvoke`. Would prefer to align on one pattern (in particular, I prefer the former, because it's nice to see the namespace).